File: src/mathlib.py
#!/usr/bin/env python3.10

# Setup search path:
import sys ; sys.path.append('target/python')

# Load SWIG bindings:
import mathlib_swig as mathlib

# Use SWIG bindings:
print(f'MATHLIB_VERSION = {mathlib.MATHLIB_VERSION}')
print(mathlib.cubic_poly(2.0, 3.0, 5.0, 7.0, 11.0))

File: src/mathlib.rb
#!/usr/bin/env ruby

# Setup search path:
ENV["LD_LIBRARY_PATH"] = 'target/ruby'
$:.unshift 'target/ruby'

# Load SWIG bindings:
require 'mathlib_swig'
include Mathlib_swig

# Use SWIG bindings:
puts "MATHLIB_VERSION = #{MATHLIB_VERSION.inspect}"
puts cubic_poly(2.0, 3.0, 5.0, 7.0, 11.0)

